app/models/concerns/annotating.rb

```diff
@@ -17,7 +17,7 @@ def models
       return @models if @models
 
       # this is to see all models in dev env, otherwise some may not be yet loaded
-      Rails.autoloaders.main.eager_load_dir("#{Rails.root}/app/models") if Rails.env.development?
+      Rails.autoloaders.main.eager_load_dir("#{Rails.root}/app/models") unless Rails.env.production?
```

Comment:
doesn't test env also use eager loading by default?

Reply:
Apparently not when the trigger is created (when running the migration or when launching `RAILS_ENV=test bundle exec rails multitenant:triggers`).

Since not all classes are loaded, the new test for the trigger was failing because the trigger didn't include `BackendAPI`

Now I'm not sure it'll work in production neither.

Reply:
I have a porta project all setup to run in production mode locally. I ran the migration:

```
SAFETY_ASURED=1 RAILS_ENV=production bundle exec rails db:migrate
```

and this is the resulting trigger:

```sql
CREATE DEFINER=`root`@`%` TRIGGER `annotations_tenant_id` BEFORE INSERT ON `annotations` FOR EACH ROW BEGIN
  DECLARE master_id numeric;
  IF @disable_triggers IS NULL THEN
    IF NEW.tenant_id IS NULL THEN
      SET master_id = 1;
      IF NEW.annotated_type = 'Account' THEN
SET NEW.tenant_id = (SELECT tenant_id FROM accounts WHERE id = NEW.annotated_id AND tenant_id <> master_id);
END IF;

    END IF;
  END IF;
END
```

😭😭😭

I'll remove the condition, better just eager load in all environments

Reply:
Fixed, I hope: https://github.com/3scale/porta/pull/3857/commits/289841036ad625214f91f0b03d6741a9d64709cb

Reply:
So eager loading will happen only when  `#models` is called. This should not happen usually  I assume :thinking: